### PR TITLE
Readme cleanup

### PR DESCRIPTION
**Pull request by cdeszaq**
Significant readme cleanup and doc love.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,69 +1,187 @@
...
-There is [a list of Key Performance Indicators (KPIs)](kpis.md) for the web service. It doesn't go into detail about what each of the KPIs mean, but it gives a rough overview of why they matter.  
+- [Metrics](docs/end-user-api.md#metrics)
+- [Dimensions](docs/end-user-api.md#dimensions)
+- [Tables](docs/end-user-api.md#tables)
+- Time ([Reporting Time Grain](docs/end-user-api.md#time-grain) and [Interval](docs/end-user-api.md#interval))
 
+The biggest difference between Fili and those other systems, however, is that Fili _only_ exposes these concepts. With
+just these core concepts, end-users can ask a huge range of questions, extracting business value by focusing on finding 
+answers to the _what_ and _why_. Other reporting systems expose end-users to other concepts like Views, Partitions,
+and metric formulas, forcing users to worry too much about _how_ to answer their questions, but not Fili.
```
Comment (cdeszaq) [inline README.md]: The intent of that final clause is to draw a sharper contrast between other systems and Fili. I don't want to lose the sharp distinction, and "Other reporting systems expose end-users to other concepts like Views, Partitions, and metric formulas, forcing users to worry too much about _how_ to answer their questions" only sounded like an unfinished thought. Suggestions?

File: README.md [hunk 2]
```diff
@@ -1,69 +1,187 @@
...
-===============================================================================================================
+Fili 
+====
 
+[![Download](https://api.bintray.com/packages/yahoo/maven/fili/images/download.svg) ](https://bintray.com/yahoo/maven/fili/_latestVersion) [![Gitter](https://img.shields.io/gitter/room/yahoo/fili.svg?maxAge=2592000)](https://gitter.im/yahoo/fili) [![Travis](https://img.shields.io/travis/yahoo/fili/master.svg?maxAge=2592000)](https://travis-ci.org/yahoo/fili/builds/) [![Codacy grade](https://img.shields.io/codacy/grade/91fa6c38f25d4ea0ae3569ee70a33e38.svg?maxAge=2592000)](https://www.codacy.com/app/Fili/fili/dashboard) [![Codacy coverage](https://img.shields.io/codacy/coverage/91fa6c38f25d4ea0ae3569ee70a33e38.svg?label=test coverage&maxAge=2592000)](https://www.codacy.com/app/Fili/fili/dashboard) [![VersionEye](https://img.shields.io/versioneye/d/user/projects/57eb1529bd6fa600512e4af1.svg?maxAge=2592000)](https://www.versioneye.com/user/projects/57eb1529bd6fa600512e4af1)
```
Comment (cdeszaq) [inline README.md]: Can you take a screenshot, and possibly look at the requests in a browser Console? Many of the badges are going through Shields.io, and so it's possible that that service is down. For me, I see everything rendering as expected.

As for the super long line, I agree it's ugly, but based on my experiments, linebreaks put things on new lines in GitHub-flavored Markdown.

An example:

oneLine
anotherLine

If we can get around that, I'm all for breaking that to more than 1 line.

Comment (archolewa) [inline README.md]: badges

Interestingly, if look at the `Network` resources under the developer console, I can see the links to get the badges (and when I open those links directly in the browser, the other badges display correctly).

Comment (archolewa) [inline README.md]: Hmm. I don't know why the `Travis` badge is showing up as broken in the screenshot.  I can see that badge fine...

Comment (cdeszaq) [inline README.md]: sssoooo... user error?

Comment (archolewa) [inline README.md]: o_O Except, I don't see it in Firefox either, and my Firefox is totally vanilla (I _am_ using a custom theme in my Chrome instance). We'll have to see when it gets merged. It's possible that there's some weird interaction going on between my browsers, the badges, and Github's preview code.

File: README.md (hunk 1 above)
Comment (archolewa) [inline README.md]: What about something like:

```markdown
On the other hand, other reporting systems expose end-users to other concepts like Views, Partitions, and metric formulars, forcing users to worry too much about _how_ to answer their questions.
```
Or replace `On the other hand` with `Meanwhile` or `In contrast` or something similar.

File: README.md [hunk 3]
```diff
@@ -1,69 +1,187 @@
...
+The most bleeding-edge version is: [![Bleeding-edge](https://api.bintray.com/packages/yahoo/maven/fili/images/download.svg) ](https://bintray.com/yahoo/maven/fili/_latestVersion)
+
+The most recent stable version is: [![Stable](https://img.shields.io/badge/Stable-0.1.17-blue.svg) ](https://bintray.com/yahoo/maven/fili/0.1.17)
+
+
+Extending
+---------
 
-The easiest way to run tests on a Mac is to use in memory dimension backend and key value store: 
+Fili's easy to extend! It has a bunch of hooks already (`AbstractBinderFactory`)! There's a module system for config and
```
Comment (archolewa) [inline README.md]: I'd suggesting making `AbstractBinderFactory` a link to that file on master.

**Comment by archolewa** [comment]
A few minor suggestions, but nothing major.

**Comment by archolewa** [comment]
👍

**Comment by sriram-viswanathan** [comment]
Looks good 👍

**Pull request merged**